app/src/main/java/pl/org/akai/movies/fragments/SignUpFragment.kt

```diff
@@ -0,0 +1,72 @@
+package pl.org.akai.movies.fragments
+
+
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.core.view.isVisible
+import com.google.firebase.auth.FirebaseAuth
+import kotlinx.android.synthetic.main.fragment_sign_up.*
+import pl.org.akai.movies.R
+
+
+class SignUpFragment : BaseFragment() {
+
+    override val layoutId: Int
+        get() = R.layout.fragment_sign_up
+
+    private lateinit var auth: FirebaseAuth
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        auth = FirebaseAuth.getInstance()
+        signUpButton.setOnClickListener {
+            val email: String = emailEditText.text.toString()
+            val password: String = passwordEditText.text.toString()
+            createUserWithEmailAndPassword(email, password)
+        }
+        auth.addAuthStateListener {
+            val user = it.currentUser
```

Comment:
lepiej to robić jako onSuccesListener w createUser, bo tak będzie mógł kilka razy wysłać nie porzebnie,
dodatkowo warto obsługiwać błędy w onFailureListenerze np. za krótkie hasło czy istniejący email